web/quickfeed_service.go

```diff
@@ -391,12 +389,12 @@ func (s *QuickFeedService) GetSubmissionsByCourse(_ context.Context, in *connect
 func (s *QuickFeedService) UpdateSubmission(_ context.Context, in *connect.Request[qf.UpdateSubmissionRequest]) (*connect.Response[qf.Void], error) {
 	if !s.isValidSubmission(in.Msg.SubmissionID) {
 		s.logger.Errorf("UpdateSubmission failed: submission author has no access to the course")
-		return nil, status.Error(codes.PermissionDenied, "submission author has no course access")
+		return nil, connect.NewError(connect.CodeInvalidArgument, errors.New("submission author has no course access"))
```

Comment:
Should this be `PermissionDenied`? It got changed to `InvalidArgument`, but in `RebuildSubmissions()` we still use `PermissionDenied`. That said, ideally, `PermissionDenied` should be handled in the interceptors, if we have sufficient information to do so.

Reply:
Yes, this is a remaining access control check I refer to in Issue #806, that should be moved to the access control interceptor. Changing it back to permission denied for now.